### PR TITLE
[FLINK-25188][python][build] Support m1 chip

### PR DESCRIPTION
**Pull request by deadwind4**
## What is the purpose of the change

Support m1 chip in PyFlink

## Brief change log

  - *Update python 3.7 3.8 dependencies: numpy>= 1.21.4, pyarrow>=5.0.0, pandas>=1.3.0, apache-beam==2.38.0, cloudpickle==2.1.0, pemja== 0.1.5*
  - *Fix Beam API to adapt 2.38.0 version*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by maver1ck** [comment]
@deadwind4 Do you know why Azure job failed ?

**Comment by deadwind4** [comment]
@maver1ck It's maybe a deserialization error, I am researching it.

**Comment by deadwind4** [comment]
@flinkbot run azure

**Comment by maver1ck** [comment]
Success :) Good work @deadwind4

File: flink-python/README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ The auto-generated Python docs can be found at [https://nightlies.apache.org/fli
 
 ## Python Requirements
 
-Apache Flink Python API depends on Py4J (currently version 0.10.9.3), CloudPickle (currently version 1.2.2), python-dateutil(currently version 2.8.0), Apache Beam (currently version 2.27.0).
+Apache Flink Python API depends on Py4J (currently version 0.10.9.3), CloudPickle (currently version 2.0.0), python-dateutil(currently version 2.8.0), Apache Beam (currently version 2.38.0).
```
Comment (maver1ck) [inline flink-python/README.md:29]: Version 2.1.0 ?

File: NOTICE [hunk 2]
```diff
@@ -17,7 +17,7 @@ See bundled license files for details.
 This project bundles the following dependencies under the BSD license.
 See bundled license files for details.
 
-- cloudpickle:1.2.2
+- cloudpickle:2.0.0
```
Comment (maver1ck) [inline NOTICE:20]: Version 2.1.0 ?

**Comment by deadwind4** [comment]
I need to make the CI auto-install different dev-requirements.txt in the python 3.6 version.

File: docs/content/docs/dev/table/sqlClient.md [hunk 3]
```diff
@@ -174,7 +174,7 @@ Mode "embedded" (default) submits Flink jobs from the local machine.
                                                --pyExecutable
                                                /usr/local/bin/python3). The python
                                                UDF worker depends on Python 3.6+,
-                                               Apache Beam (version == 2.27.0), Pip
+                                               Apache Beam (version == 2.38.0), Pip
```
Comment (HuangXingBo) [inline docs/content/docs/dev/table/sqlClient.md:177]: Does the version of pip need to be changed?

**Review by HuangXingBo — COMMENTED** [review]
@deadwind4 Thanks a lot for the PR. Looks good overall. I only have one comment?

File: docs/content/docs/dev/table/sqlClient.md (hunk 3 above)
Comment (deadwind4) [reply docs/content/docs/dev/table/sqlClient.md:177]: Should I also add `pip>=20.3` into `dev-requirements.txt` ?

**Review by HuangXingBo — APPROVED** [review]

**Pull request closed without merging**